File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)

{
    long card = get_long("Number: "); //Get CC nr.
    long cardUntouched = card;


    int otherDigit = 0;
    int beforeLastDigit = 0;
    int sum = 0;
    int sumSecond = 0;
    //Check the checksum to validate if card is a valid cc nr.
    while (card > 10)
    {
        beforeLastDigit = (card % 10); // Get the last digit.
        card = (card - beforeLastDigit) / 10; // Remove the last digit from the card.

        otherDigit += beforeLastDigit; // Get the digits starting with the last one added together.

        beforeLastDigit = (card % 10);
        card = (card - beforeLastDigit) / 10; // Get the second to last digit

        sumSecond = beforeLastDigit * 2; //Add digit * 2 to see later if it is < 1;0

        if (beforeLastDigit < 10 && sumSecond < 10) // Check if digit * 2 > 10.
        {
            sum += beforeLastDigit * 2;
        }
        else
        {
            int sumE = sumSecond;
            sumE = (sumSecond % 10);
            int sumX = (sumSecond - sumE) / 10;  // If sum > 10 then get digits one by one for ex. if sum = 13 => 1+3.
            sum = sum + sumX + sumE;
        }

    }


    if (card < 10 && card > 0)
    {
        otherDigit += card;
        card = card - card;
    }


    int total = sum + otherDigit;
    if (total % 10 == 0) // Check is is a valid CC nr.
    {
        long int cardcopy  = cardUntouched;
        long int cardcopytwo = cardUntouched;
        int i;

        for (i = 0; cardcopy > 0; i++) // find the number of digits
        {

            cardcopy = (cardcopy - (cardcopy % 10)) / 10; // Remove the last digit from the card.

        }
        while (cardcopytwo > 99)
        {
            cardcopytwo = (cardcopytwo - (cardcopytwo % 10)) / 10; // Get the first 2 digits in cardcopy variable

        }


        if (i == 15 && (cardcopytwo == 34 || cardcopytwo == 37)) // i = nr of digits cardcopy = first 2 digits
        {
            printf("AMEX\n");
        }
        else if (i == 16 && (cardcopytwo <= 55 && cardcopytwo >= 51)) // i = nr of digits cardcopy = first 2 digits
        {
            printf("MASTERCARD\n");
        }
        else if ((i == 16 || i == 13) && (((cardcopytwo - (cardcopytwo % 10)) / 10) == 4)) // i = nr of digits cardcopy = first 2 digits
        {

            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }

    }
    else
    {
        printf("INVALID\n");
    }

}

File: pset6/credit/credit.py
from cs50 import get_string

card = get_string("Number: ") #Get CC Nr.

reverseCard = [int(x) for x in card[::-1]]
listDouble = list()
digits = list(enumerate(reverseCard, start=1))

for index, digit in digits:
    if index % 2 == 0:
        dDigit = digit * 2
        if dDigit > 9:
            listDouble.append(dDigit - 9)
        else:
            listDouble.append(dDigit)
    else:
        listDouble.append(digit)

    cardLen = [13, 15, 16]

if not sum(listDouble) % 10 == 0 or len(card) not in cardLen:
    print("INVALID")
    exit(0)

amex = ['34', '37']
mastercard = ['51', '52', '53', '54', '55']

if card[:2] in amex and len(card) == 15:
    print("AMEX")
elif card[:2] in mastercard and len(card) == 16:
    print("MASTERCARD")
elif len(card) == 16 and card[:1] == '4' or len(card) == 13 and card[:1] == '4':
    print("VISA")
else:
    print("INVALID")




